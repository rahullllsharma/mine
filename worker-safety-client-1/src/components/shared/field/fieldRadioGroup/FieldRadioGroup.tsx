import type { RadioGroupProps } from "../../radioGroup/RadioGroup";
import type { FieldProps } from "../Field";
import Paragraph from "../../paragraph/Paragraph";
import RadioGroup from "../../radioGroup/RadioGroup";
import Field from "../Field";

export type FieldRadioGroupProps = FieldProps &
  RadioGroupProps & {
    readOnly?: boolean;
  };

export default function FieldRadioGroup({
  options,
  defaultOption,
  onSelect,
  hasError,
  readOnly,
  ...fieldProps
}: FieldRadioGroupProps): JSX.Element {
  return (
    <Field {...fieldProps}>
      {readOnly ? (
        <Paragraph
          text={defaultOption?.description || defaultOption?.value.toString()}
        />
      ) : (
        <RadioGroup
          options={options}
          defaultOption={defaultOption}
          hasError={hasError}
          onSelect={onSelect}
        />
      )}
    </Field>
  );
}
