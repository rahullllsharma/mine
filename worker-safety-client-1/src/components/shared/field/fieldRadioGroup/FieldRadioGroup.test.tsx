import type { FieldRadioGroupProps } from "./FieldRadioGroup";
import { render } from "@testing-library/react";
import FieldRadioGroup from "./FieldRadioGroup";

const DUMMY_OPTIONS: FieldRadioGroupProps["options"] = [
  { id: 1, value: "One" },
  { id: 2, value: "Two" },
  { id: 3, value: "Three" },
];

describe(FieldRadioGroup.name, () => {
  it('should render with the "RadioGroup" component', () => {
    const { asFragment } = render(
      <FieldRadioGroup
        label="Lorem ipsum"
        options={DUMMY_OPTIONS}
        defaultOption={{ id: 3, value: "Three" }}
        onSelect={jest.fn()}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('should render a paragraph with the default option when the "readOnly" is set', () => {
    const { asFragment } = render(
      <FieldRadioGroup
        label="Lorem ipsum"
        options={DUMMY_OPTIONS}
        onSelect={jest.fn()}
        defaultOption={DUMMY_OPTIONS[1]}
        readOnly
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
