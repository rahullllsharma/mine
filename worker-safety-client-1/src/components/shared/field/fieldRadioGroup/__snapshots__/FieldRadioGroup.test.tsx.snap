// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldRadioGroup should render a paragraph with the default option when the "readOnly" is set 1`] = `
<DocumentFragment>
  <div>
    <label
      class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
    >
      Lorem ipsum
    </label>
    <p
      class="text-base text-neutral-shade-100"
    >
      Two
    </p>
  </div>
</DocumentFragment>
`;

exports[`FieldRadioGroup should render with the "RadioGroup" component 1`] = `
<DocumentFragment>
  <div>
    <label
      class="block text-tiny md:text-sm text-neutral-shade-75 font-semibold mb-2 leading-normal"
    >
      Lorem ipsum
    </label>
    <div
      class="flex flex-wrap flex-row gap-6"
      id="headlessui-radiogroup-7"
      role="radiogroup"
    >
      <div
        aria-checked="false"
        aria-labelledby="headlessui-label-1"
        class="flex items-center"
        id="headlessui-radiogroup-option-2"
        role="radio"
        tabindex="-1"
      >
        <div
          class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
        />
        <label
          class="text-neutral-shade-100 text-base ml-2"
          id="headlessui-label-1"
        >
          One
        </label>
      </div>
      <div
        aria-checked="false"
        aria-labelledby="headlessui-label-3"
        class="flex items-center"
        id="headlessui-radiogroup-option-4"
        role="radio"
        tabindex="-1"
      >
        <div
          class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
        />
        <label
          class="text-neutral-shade-100 text-base ml-2"
          id="headlessui-label-3"
        >
          Two
        </label>
      </div>
      <div
        aria-checked="true"
        aria-labelledby="headlessui-label-5"
        class="flex items-center"
        id="headlessui-radiogroup-option-6"
        role="radio"
        tabindex="0"
      >
        <div
          class="flex items-center justify-center border rounded-xl w-5 h-5 border-neutral-shade-38 cursor-pointer hover:border-neutral-shade-58 hover:shadow-10"
        >
          <div
            class="w-2.5 h-2.5 rounded-xl bg-brand-urbint-40 shadow-5"
          />
        </div>
        <label
          class="text-neutral-shade-100 text-base ml-2"
          id="headlessui-label-5"
        >
          Three
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
